src/canmatrix/canmatrix.py

```diff
@@ -551,6 +552,48 @@ def pack_bitstring(length, is_float, value, signed):
 
     return bitstring
 
+@attr.s(cmp=False)
+class Arbitration_Id(object):
+    standard_id_mask = ((1 << 11) - 1)
+    extended_id_mask = ((1 << 29) - 1)
+    compound_extended_mask = (1 << 31)
+
+    id = attr.ib(default=None)
+    extended = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        if self.extended is None or self.extended:
+            mask = self.extended_id_mask
+        else:
+            mask = self.standard_id_mask
+
+        if self.id != self.id & mask:
+            raise ArbitrationIdOutOfRange('ID out of range')
+
+    @classmethod
+    def from_compound_integer(cls, i):
+        return cls(
+            id=i & cls.extended_id_mask,
+            extended=(i & cls.compound_extended_mask) != 0,
+        )
+
+    def to_compound_integer(self):
+        if self.extended:
+            return self.id | self.compound_extended_mask
+        else:
+            return self.id
+
+    def __eq__(self, other):
```

Comment:
I think we should be able to drop this after the above changes?  The default attrs-provided `__eq__` should work and if it doesn't we should probably try to look for a way to make it.